### PR TITLE
Cloudberry DB initial support

### PR DESCRIPTION
**Pull request by ostinru**
CloudberryDB initial support:
 * Introduce greenplum.Flavor
 * add it to sentinel
 * update SQL queries to work with Cloudberry
 * copy all Greenplum docker tests to Cloudberry tests (and fix it)

### CODE REVIEW

File: docker/cloudberry_tests/scripts/tests/backup_after_failed_backup_test.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# show the backup list
+wal-g backup-list --config=${TMP_CONFIG}
+
+# show the storage objects (useful for debug)
+wal-g st ls -r --config=${TMP_CONFIG}
+
+# it looks like a bug in `tryExtractFiles` - it can override files when WALG_DOWNLOAD_CONCURRENCY != 1
+# https://github.com/wal-g/wal-g/blob/master/internal/extract.go#L201-L253
+WALG_DOWNLOAD_CONCURRENCY=1  wal-g backup-fetch LATEST --in-place --config=${TMP_CONFIG}
```
Comment (ostinru) [inline docker/cloudberry_tests/scripts/tests/backup_after_failed_backup_test.sh:36]: Here we have an issue with 'backup_label's:

in `part_001.tar.lz4` something like
```
START WAL LOCATION: 0/C000028 (file 000000010000000000000003)
CHECKPOINT LOCATION: 0/C000060
BACKUP METHOD: pg_start_backup
BACKUP FROM: primary
START TIME: 2024-10-15 19:38:24 UTC
LABEL: abc
START TIMELINE: 1
```

in `part_003.tar.lz4` something like
```
START WAL LOCATION: 0/C000028 (file 000000010000000000000003)
CHECKPOINT LOCATION: 0/C000060
BACKUP METHOD: streamed
BACKUP FROM: primary
START TIME: 2024-10-15 19:46:14 UTC
LABEL: 2024-10-15 19:46:13.608579 +0000 UTC m=+0.212280844
START TIMELINE: 1
```

as a result we getting
```
requested recovery stop point is before consistent recovery point
```

File: docker/cloudberry_tests/scripts/tests/test_functions/util.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      more "${arr[1]}"/log/* | cat || true
+    done
+    exit 1
+}
+
+stop_cluster() {
+  /usr/local/gpdb_src/bin/gpstop -a -M fast
+}
+
+prepare_cluster() {
```
Comment (ostinru) [inline docker/cloudberry_tests/scripts/tests/test_functions/util.sh:64]: difference from GP v6:

File: docker/cloudberry/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    zlib1g-dev \
+    libpstreams-dev \
+    libxerces-c-dev
+
+ADD docker/cloudberry/run_greenplum.sh /home/gpadmin/run_greenplum.sh
+
+WORKDIR /usr/local
+# FIXME: checkout main barnch... however it doesn't compile: https://github.com/cloudberrydb/cloudberrydb/pull/611#issuecomment-2413911473
+RUN git clone https://github.com/cloudberrydb/cloudberrydb.git gpdb_src --single-branch --branch 1.6.0 --depth 1 \
+ && ./gpdb_src/concourse/scripts/setup_gpadmin_user.bash
```
[the remark this answers is not in the document]
Comment (reshke) [reply docker/cloudberry/Dockerfile:72]: > [cloudberrydb/cloudberrydb#611 (comment)](https://github.com/cloudberrydb/cloudberrydb/pull/611#issuecomment-2413911473)

https://github.com/cloudberrydb/cloudberrydb/pull/674 ?

[the remark this answers is not in the document]
Comment (ostinru) [reply docker/cloudberry/Dockerfile:71]: it looks like https://github.com/cloudberrydb/cloudberrydb/pull/673 fixed compillation issues in 'main' branch

**Review by debebantur — APPROVED** [review]

**Pull request merged**